src/reporter.cc

```diff
@@ -71,46 +78,109 @@ void MetadataReporter::ReportMetadata() {
 namespace {
 
 void SendMetadataRequest(std::vector<json::value>&& entries,
-                         const std::string& endpoint,
+                         const std::string& batch_uri,
+                         const std::string& publish_path,
                          const std::string& auth_header,
                          const std::string& user_agent,
                          bool verbose_logging)
     throw (boost::system::system_error) {
-  json::value update_metadata_request = json::object({
-    {"entries", json::array(std::move(entries))},
-  });
 
-  if (verbose_logging) {
-    LOG(INFO) << "About to send request: POST " << endpoint
-              << " User-Agent: " << user_agent
-              << " " << *update_metadata_request;
+  if (entries.size() == 1) {
+    // Add a copy of the single entry, except for the `views` field. This allows
+    // us to sent a request to the batch endpoint when we have a single request.
+    const json::Object* single_request = entries[0]->As<json::Object>();
+    json::value duplicate_request = json::object({  // ResourceMetadata
```

Comment:
I wonder if this would be more readable as an exact implementation of your comment above, e.g., something like:
```c++
    json::value duplicate_request = entries[0]->Clone();
    json::Object* request_object = duplicate_request->AsMutable<json::Object>();
    request_object->erase("views");
    entries.emplace_back(std::move(duplicate_request));
```
(We would need to rename `json::Value::As` to `json::Value::AsMutable` that returns a non-const pointer, and then have `json::Value::As` call `json::Value::AsMutable`)...